osp/core/session/session.py

```diff
@@ -18,13 +18,19 @@ def __init__(self):
         self._registry = Registry()
         self.root = None
         self.graph = rdflib.Graph()
+        self._previous_session = None
 
     def __enter__(self):
         """Establish the connection to the backend."""
+        from osp.core.cuds import Cuds
```

Comment:
What would happen if you had nested `with` statements?
Wouldn't this always set the previous to `CoreSession`?

Reply:
You're right. The previous session should be a member of Cuds I guess

Reply:
I guess unittests should be added to this

Reply:
okay you're not right

Reply:
In any case, we need an unit test for this. Maybe I can write it , that would make the code review more efective.